src/interfaces/node.cpp

```diff
@@ -83,8 +83,8 @@ class NodeImpl : public Node
     bool shutdownRequested() override { return ShutdownRequested(); }
     void mapPort(bool use_upnp) override
     {
-        if (use_upnp) {
-            StartMapPort();
+        if (use_upnp && g_connman) {
```

Comment:
In commit "net: pass port into MapPort/Discover rather than using globals" (4d0cac7be150145ec45119188458a9625f882b29)

I think it instead of adding `&& g_connman` here, it would make more sense to `assert(g_connman)`, or just do nothing if `g_connman` is null. Falling back to the else condition and having a `mapPort(use_upnp=true)` trigger `StopMapPort` in this case doesn't seem like it could be helpful.